### PR TITLE
Issue #418: automate execution by Github action bump-license-year.sh

### LINKED ISSUE #418: automate execution by Github action bump-license-year.sh on first day of month
First day of 1st month of the year:
`0 0 1 1 *`

This simple task will let us have the experience to update sources from Action, which we will need during release notes update in sources.

### PR DESCRIPTION
**Pull request by Rahulkhinchi03**
Fixes #418

### CODE REVIEW

**Comment by muhlba91** [comment]
can you please do the following:
- pass in year 2022 and 2023
- list all changed files (e.g. git add --all / git status)
- attach this list to this PR as a proof of changes

File: .github/workflows/bump-license-year.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    steps:
+      - name: Checkout the latest code
+        uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+      - name: Set Current Year
+        id: CURRENT_YEAR
+        run: |
+          echo "::set-output name=year::$(date +'%Y')"
+      - name: Modify File
```
Comment (romani) [inline .github/workflows/bump-license-year.yml:27]: Files

**Review by romani — CHANGES_REQUESTED** [review]
Apply same change for all repos

Items

File: .github/workflows/bump-license-year.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          ./.ci/bump-license-year.sh $(expr ${{ env.YEAR }} - 1) ${{ env.YEAR }} .
+        env:
+          YEAR: ${{ steps.CURRENT_YEAR.outputs.year }}
+      - name: Push commit
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git commit -am "minor: Bump year to ${{ env.YEAR }}"
+          git push
+
```
[the remark this answers is not in the document]
Comment (Rahulkhinchi03) [reply .github/workflows/bump-license-year.yml]: Changes Done in all Repos.

File: .github/workflows/bump-license-year.yml (hunk 1 above)
Comment (Rahulkhinchi03) [reply .github/workflows/bump-license-year.yml]: Changes done in all repos.

**Comment by Rahulkhinchi03** [comment]
> can you please do the following:
> 
> * pass in year 2022 and 2023
> * list all changed files (e.g. git add --all / git status)
> * attach this list to this PR as a proof of changes
> 
> otherwise, LGTM

working on it.

File: .github/workflows/bump-license-year.yml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        run: |
+          ./.ci/bump-license-year.sh $(expr ${{ env.YEAR }} - 1) ${{ env.YEAR }} .
+        env:
+          YEAR: ${{ steps.CURRENT_YEAR.outputs.year }}
+      - name: Push commit
+        run: |
+          git config --global user.name 'github-actions[bot]'
+          git config --global user.email 'github-actions[bot]@users.noreply.github.com'
+          git commit -am "minor: Bump year to ${{ env.YEAR }}"
+          git push
```
Comment (romani) [inline .github/workflows/bump-license-year.yml:37]: Add /n

**Review by romani — CHANGES_REQUESTED** [review]
Items

File: .github/workflows/bump-license-year.yml (hunk 3 above)
Comment (Rahulkhinchi03) [reply .github/workflows/bump-license-year.yml:37]: At first, there was /n already. Do you want me to revert back to https://github.com/checkstyle/sonar-checkstyle/compare/e9f34296a8ad2d417e49b2f0e03ec2c9c47c7528..1e2351f72a6d24f311e7b7786efe8423494dd212 ?

Comment (romani) [reply .github/workflows/bump-license-year.yml:37]: Please get rid of red sign that GitHub shows you

Comment (Rahulkhinchi03) [reply .github/workflows/bump-license-year.yml]: Changes Done

**Review by romani — APPROVED** [review]
ok to merge, as request for tests from @muhlba91 is satisfied

**Comment by Rahulkhinchi03** [comment]
@muhlba91 @romani 
Commit of file for testing `2023 year` : https://github.com/checkstyle/sonar-checkstyle/commit/622d87039e06d8b45de59f476cfadccad11eb6e0

Information on all the changes (please check push changes): https://github.com/Rahulkhinchi03/sonar-checkstyle/runs/7372408589?check_suite_focus=true

**Review by muhlba91 — APPROVED** [review]

**Pull request merged**